probe-rs-target/src/chip.rs

```diff
@@ -1,6 +1,42 @@
 use super::memory::MemoryRegion;
 use crate::{serialize::hex_option, CoreType};
 use serde::{Deserialize, Serialize};
+
+/// Represents a DAP scan chain element.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ScanChainDap {
+    /// Unique name of the DAP
+    pub name: Option<String>,
+    /// Specifies the type of the DAP (e.g. ARMCS-DP)
+    pub dap_type: Option<String>,
+    /// Specifies the IR length of the DAP (default value: 4).
+    pub ir_len: Option<u8>,
+    /// List of protocols via which the DAP can be use. Possible values are SWD, JTAG, and cJTAG.
+    pub protocols: Option<Vec<String>>,
+    /// Specifies the SW-DPv2 TARGETSEL register value that selects this DAP in a serial-wire debug multi-drop system
+    pub targetsel: Option<u8>,
+}
+
+/// Represents a non DAP device in the scan chain. This can be other TAPs.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct ScanChainDevice {
+    /// Unique name of the device
+    pub name: Option<String>,
+    /// Specifies the type of the device (e.g. ARMCS-DP)
+    pub device_type: Option<String>,
+    /// Specifies the IR length of the device (default value: 4).
+    pub ir_len: Option<u8>,
+}
+
+/// Declares the type of a scan chain as defined by CMSIS-Pack SDF
+/// ref: https://open-cmsis-pack.github.io/Open-CMSIS-Pack-Spec/main/html/sdf_pg.html#sdf_element_scanchain
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub enum ScanChain {
+    /// Scan chain element is a DAP
+    Dap(ScanChainDap),
```

Comment:
Maybe DAPs should actually be stored in a device and we don't need this enum?

Reply:
You're right, that was just over complicating things. Removed.